File: 2025/5/5/p1.cs
// var lines = File.ReadAllLines("../../../input.txt").ToList();
//
//
// List<(long, long)> Ranges = new List<(long, long)>();
// foreach (var line in lines)
// {
//     if (line.Trim() == "")
//     {
//         break;
//     }
//
//     var Parts = line.Split("-").ToList().Select((s => Int64.Parse(s))).ToList();
//     Ranges.Add((Parts[0],Parts[1]));
// }
//
// int ok = 0;
// foreach (var line in lines)
// {
//     if (line.Contains("-") || line.Trim() == "")
//     {
//         continue;
//     }
//     foreach (var range in Ranges)
//     {
//         var number = Int64.Parse(line);
//         if (number <= range.Item2 && number >= range.Item1)
//         {
//             ok++;
//             break;
//         }
//     }
// }
//
// Console.WriteLine(ok);

File: 2025/7/p1.py
with open("input.txt", 'r') as f:
    lines = [list(line) for line in f.readlines()]

splits = 0
for i in range(1, len(lines)):
    for c in range(len(lines[i])):
        if lines[i-1][c] == 'S':
            lines[i][c] = '|'
        if lines[i-1][c] == '|':
            if lines[i][c] == '^':
                splits+=1
                lines[i][c-1] = '|'
                lines[i][c+1] = '|'
            else:
                lines[i][c] = '|'
print(splits)